### PR TITLE
Network: Remove NodeDIDResolver

### LINKED ISSUE #2015: Network: Node DID must be set at startup / remove NodeDIDResolver concept
If the node DID is not available at startup, the private transaction request loop is not started.

Currently the `autoNodeDIDResolver` can pickup on new DIDs at any point (in non-strict mode). If this happens after startup, it would appear that a node is able to receive private transactions based on all diagnostics info, but still fails due to the request loop not being started.

Solution is to only auto discover DID once at startup. This is just developer convenience and should still not happen in strict mode. This also allows removal of the `NodeDIDResolver` concept to simplify code.

### PR DESCRIPTION
**Pull request by gerardsn**
closes #2015 
the NodeDID is no longer discovered on a running node. It has to be configured, or detected at startup.

### CODE REVIEW

**Comment by gerardsn** [comment]
Linked pull request fixes the E2E tests

**Review by woutslakhorst — APPROVED** [review]
e2e PR is also approved. Merge that first.

File: network/transport/nodedid.go [hunk 1]
```diff
@@ -24,71 +24,29 @@ import (
...
-func (a *autoNodeDIDResolver) Resolve(ctx context.Context) (did.DID, error) {
-	a.mux.Lock()
-	defer a.mux.Unlock()
-	if !a.resolvedDID.Empty() {
-		result := a.resolvedDID
-		return result, nil
-	}
-
-	documents, err := a.docFinder.Find(didservice.IsActive(), didservice.ValidAt(time.Now()), didservice.ByServiceType(NutsCommServiceType))
+func AutoResolveNodeDID(ctx context.Context, keyResolver crypto.KeyResolver, docFinder types.DocFinder) (*did.DID, error) {
```
Comment (reinkrul) [inline network/transport/nodedid.go:31]: if `*did.DID` is never nil, just return `did.DID`

File: network/transport/grpc/connection_manager.go [hunk 2]
```diff
@@ -129,7 +129,7 @@ type grpcConnectionManager struct {
 	ctxCancel           func()
 	listener            net.Listener
 	authenticator       Authenticator
-	nodeDIDResolver     transport.NodeDIDResolver
+	nodeDID             *did.DID
```
[the remark this answers is not in the document]
Comment (gerardsn) [reply network/transport/grpc/connection_manager.go:132]: An empty DID is set in the constructor. I prefer to have a pointer so the network / protocol / connection manager all point to the same DID.

[the remark this answers is not in the document]
Comment (reinkrul) [reply network/transport/grpc/connection_manager.go:132]: These are created after the node DID is resolved, so you can just pass it?

**Pull request merged**